osbrain/tests/test_agent_sync_publications.py

```diff
@@ -0,0 +1,238 @@
+"""
+Test file for synchronized publications.
+"""
+import time
+
+import pytest
+
+from osbrain import Agent
+from osbrain import run_agent
+from osbrain import run_logger
+
+from common import nsproxy  # pragma: no flakes
+from common import sync_agent_logger
+from common import logger_received
+
+
+class Server(Agent):
+    def on_init(self):
+        self.count = 0
+        self.received = []
+        self.bind('SYNC_PUB', alias='publish', handler='reply')
+
+    def publish(self):
+        self.count += 1
+        self.send('publish', self.count)
+
+    def publish_str(self):
+        self.count += 1
+        self.send('publish', str(self.count), topic='positive')
+        self.send('publish', str(-self.count), topic='negative')
+
+    def reply(self, request):
+        self.received.append((self.count, request))
+        return self.count + 0.5
+
+
+class ServerYield(Server):
+    def reply(self, request):
+        yield self.count + 0.5
+        self.received.append((self.count, request))
+
+
+class ServerLate(Server):
+    def reply(self, delay):
+        self.received.append(delay)
+        time.sleep(delay)
+        return 'x' + str(delay)
+
+
+class Client(Agent):
+    def on_init(self):
+        self.received = []
+
+
+def wait_agent_list(agent, name='received', length=None, data=None,
+                    timeout=3):
+    t0 = time.time()
+    while True:
+        received = agent.get_attr('received')
+        if length is not None and len(received) >= length:
+            return True
+        if data is not None and data in received:
+            return True
+        if time.time() - t0 > timeout:
+            break
+        time.sleep(0.01)
+    return False
+
+
+def receive(agent, response):
+    agent.received.append(response)
+
+
+def receive_times2(agent, response):
+    agent.received.append(response * 2)
```

Comment:
Well, I added a comment before and now I can't mark it as approved. But definetely the syntax seems really good to me, so feel free to go ahead with the merge  @Peque :+1: